drivers: spi_nrfx_spim: Incorrect handling of extended SPIM configuration

**Describe the bug**
Extended SPIM configuration (needed by the `SPI_3_NRF_RX_DELAY` Kconfig option) is not handled correctly in `spi_nrfx_spim.c`. In result, no other SPIM instance can be enabled together with SPIM3.

**To Reproduce**
1. In the `hello_world` sample, add the following lines to `prj.conf`:
```
CONFIG_SPI=y
CONFIG_SPI_2=y
CONFIG_SPI_2_NRF_SPIM=y
CONFIG_SPI_3=y
```
and create a file named `nrf52840_pca10056.overlay` with the following content:
```
&spi2 {
	compatible = "nordic,nrf-spim";
};
```
2. Try to build the sample for the `nrf52840_pca10056` board:
```
west build -b nrf52840_pca10056
```
3. See that compilation fails.

**Expected behavior**
Successful build.

**Impact**
showstopper

**Screenshots or console output**
```
spi_nrfx_spim.c:365:14: error: 'CONFIG_SPI_2_NRF_RX_DELAY' undeclared here (not in a function); did you mean 'CONFIG_SPI_3_NRF_RX_DELAY'?
  .rx_delay = CONFIG_SPI_##idx##_NRF_RX_DELAY,
              ^~~~~~~~~~~
```

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain: Zephyr SDK
 - Zephyr 2.1.0 rc2 (92e017fd70b203c751c1f1d7e8dc5dc4d85a1939) but the same error will occur for Zephyr 1.14.1 (77c455caa5174346e1b38dd5c5bd4bda38354e56)

--- a/drivers/spi/spi_nrfx_spim.c
+++ b/drivers/spi/spi_nrfx_spim.c
@@ -360,12 +360,12 @@ static int spim_nrfx_pm_control(struct device *dev, u32_t ctrl_command,
 }
 #endif /* CONFIG_DEVICE_POWER_MANAGEMENT */
 
-#if NRFX_CHECK(NRFX_SPIM_EXTENDED_ENABLED)
-#define SPI_NRFX_SPIM_EXTENDED_CONFIG(idx) \
-	.rx_delay = CONFIG_SPI_##idx##_NRF_RX_DELAY,
-#else
-#define SPI_NRFX_SPIM_EXTENDED_CONFIG(idx)
-#endif
+#define SPI_NRFX_SPIM_EXTENDED_CONFIG(idx)				\
+	COND_CODE_1(IS_ENABLED(NRFX_SPIM_EXTENDED_ENABLED),		\
+		(COND_CODE_1(SPIM##idx##_FEATURE_RXDELAY_PRESENT,	\
+			(.rx_delay = CONFIG_SPI_##idx##_NRF_RX_DELAY,),	\
+			())),						\
+		())
 
 #define SPI_NRFX_SPIM_DEVICE(idx)					       \
 	static int spi_##idx##_init(struct device *dev)			       \

